src/Build/BackEnd/Components/Caching/ConfigCache.cs

```diff
@@ -188,16 +188,19 @@ public bool HasConfiguration(int configId)
         /// </summary>
         public void ClearConfigurations()
         {
-            if (_configurations != null)
+            lock (_configurations)
```

Comment:
Really I think the null check shouldn't even be there. It could be a count and if 0 then return?